packages/fiori/src/DynamicSideContent.js

```diff
@@ -0,0 +1,546 @@
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import SideContentPosition from "./types/SideContentPosition.js";
+import SideContentVisibility from "./types/SideContentVisibility.js";
+import SideContentFallDown from "./types/SideContentFallDown.js";
+import DynamicSideContentTemplate from "./generated/templates/DynamicSideContentTemplate.lit.js";
+
+// Styles
+import DynamicSideContentCss from "./generated/themes/DynamicSideContent.css.js";
+
+// Texts
+import {
+	DSC_SIDE_ARIA_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-dynamic-side-content",
+	managedSlots: true,
+	properties: /** @lends sap.ui.webcomponents.main.DynamicSideContent.prototype */ {
+
+		/**
+		 * Determines whether to hide the main content.
+		 *
+		 * @type {boolean}
+		 * @public
+		 *
+		 */
+		 hideMainContent: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines whether to hide the side content.
+		 *
+		 * <b>Note:</b> If both <code>hideSideContent</code> and <code>hideMainContent</code>
+		 * properties are set to <code>false</code>, use the <code>toggleContents</code> method
+		 * for showing the side content on phone.
+		 *
+		 * @type {boolean}
+		 * @public
+		 *
+		 */
+		hideSideContent: {
+			type: Boolean,
+		},
+
+		/**
+		 * Determines whether the side content is on the left or on the right side of the main content.
+		 *
+		 * @type {SideContentPosition}
+		 * @defaultvalue "End"
+		 * @public
+		 *
+		 */
+		 sideContentPosition: {
+			type: SideContentPosition,
+			defaultValue: SideContentPosition.End,
+		},
+
+		/**
+		 * Determines on which breakpoints the side content is visible.
+		 *
+		 * @type {SideContentVisibility}
+		 * @defaultvalue "ShowAboveS"
+		 * @public
+		 *
+		 */
+		sideContentVisibility: {
+			type: SideContentVisibility,
+			defaultValue: SideContentVisibility.ShowAboveS,
+		},
+
+		/**
+		 * Determines on which breakpoints the side content falls down below the main content.
+		 *
+		 * @type {SideContentFallDown}
+		 * @defaultvalue "OnMinimumWidth"
+		 * @public
+		 *
+		 */
+		sideContentFallDown: {
+			type: SideContentFallDown,
+			defaultValue: SideContentFallDown.OnMinimumWidth,
+		},
+
+		/**
+		 * Defines whether the control is in equal split mode. In this mode, the side and
+		 * the main content take 50:50 percent of the container on all screen sizes
+		 * except for phone, where the main and side contents are switching visibility
+		 * using the toggle method.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 *
+		 */
+		equalSplit: {
+			type: Boolean,
+		},
+
+		/**
+		 * If set to TRUE, then not the media Query (device screen size) but the size of
+		 * the container, surrounding the control, defines the current range.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 *
+		 */
+		containerQuery: {
```

Comment:
I think we should remove this property. The component's look and behavior should be determined by its own size (in this  case surrounding container's size) at all times. This needs to be at least discussed.

Reply:
We had discussion on this and decided that this property must be removed.